src/parser/jobs/rpr/modules/ImmortalSacrifice.tsx

```diff
@@ -0,0 +1,154 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import React from 'react'
+import {DataLink} from '../../../../components/ui/DbLink'
+import {Event, Events} from '../../../../event'
+import {Analyser} from '../../../core/Analyser'
+import {filter} from '../../../core/filter'
+import {dependency} from '../../../core/Injectable'
+import BrokenLog from '../../../core/modules/BrokenLog'
+import Checklist, {Requirement, Rule} from '../../../core/modules/Checklist'
+import {Data} from '../../../core/modules/Data'
+import {DISPLAY_ORDER} from './DISPLAY_ORDER'
+
+interface ImmortalSacrificeWindow {
+	startTime: number,
+	endTime?: number,
+	/**
+	 * If this window was closed, by consuming the stacks, or it expired
+	 */
+	closed: boolean,
+	stackCount: number,
+	consumed: boolean,
+}
```

Comment:
Is this really necessary? The times don't matter, neither does stack count of a window doesn't matter (which isn't even used), it's simpler to just count stacks gained vs used. If you want to use a window in some sort of output chart with timeline links then it makes more sense but in that case stick to consistent field names with the rest of the codebase. Consumed probably shouldn't be a boolean either.

Reply:
Initially, I was going to offer more than just stack generated and consumed, I thought maybe checking the number of stacks you got and give a minor hint if you get low stacks, since they don't depend on you and are random, but in my experience, if everyone is keeping their gcd rolling and you hit all members with buff you should get 6+ more often then not, also would show the window for you to check when that happened what was going on, that's why I implemented this as a window at first. Later after thinking a bit about it, I'm not sure how useful this feedback to the user would be so I went for a simple generated/consumed but left the logic there to explore as I check more parses, etc.

Regarding the `consumed` I think it should be a boolean because you can't control the stack generation and when you are generating stacks you can't consume them, and when you are ready to consume you can't generate more, basically you can only consume all the stacks, so the amount consumed didn't make much sense to me, this ties to the same thing I was mentioning before, not sure how useful this info will be. I think it's easier to read something like you had the opportunity to use PH 5 times, but you only used 4 times, instead of you had 35 stacks but consumed only 28. Maybe it would make more sense to just rewrite the messages along these lines, focusing on the PH usage instead of stack.

What are you thoughts?

Reply:
Regarding the window, I think it depends on the intent. Purely for a checklist, counting stacks gained vs spent is more efficient. It's simpler to implement and more accurate. If you wanted some sort of chart with each use of Arcane Crest showing number of people hit, stacks gained, stacks spent, and a link to the timeline so they can see the rest of their rotation then I think the window has a lot more value.

I think there's value in tracking actual consumptions, but you're probably right that the core thing here is the description text. We can adjust the message to reflect the intent, we just need to decide what we want to be showing.

Reply:
Great, so I'll rework the code a bit to simplify it, but will go with the PH usage message, I think it works much better as a checklist.

Reply:
Sounds good to me! There's no reason we can't have more than one display tho - if a checklist+table is useful, we can do both, and it doesn't have to be in this PR, can always add more features later.